transport-msg.tex

```diff
@@ -940,33 +971,56 @@ \subsubsection{Overview}
 & 20 & 8 & Driver address \\
 & 28 & 8 & Device address \\
 \hline
-Answer & 0 & 4 & Virtqueue index \\
-& 4 & 4 & Reserved (Must Be Zero - MBZ) \\
-& 8 & 4 & Current virtqueue size \\
-& 12 & 8 & Descriptor address \\
-& 20 & 8 & Driver address \\
-& 28 & 8 & Device address \\
+Answer & 0 & 0 & no extra data \\
 \hline
 \end{tabular}
 
 \msgdef{RESET_VQUEUE}
 
-This message is sent by the virtio-msg transport frontend and requires a
-response from the backend.
+This message is sent by the virtio-msg transport driver and requires a
+response from the device.
+
+The driver \emph{SHOULD NOT} send this message unless the VIRTIO_F_RING_RESET
+feature has been negotiated.
 
 \begin{tabular}{|l|l|l|l|}
 \hline
 Type & Offset & Size (bytes) & Content \\
 \hline \hline
 Request & 0 & 4 & Virtqueue index \\
 \hline
-Answer & 0 & 0 & None \\
+Answer & 0 & 0 & no extra data \\
+\hline
+\end{tabular}
+
+\msgdef{EVENT_CONFIG0}
+
+This message is sent by the virtio-msg device and there is no response.
+
+\begin{tabular}{|l|l|l|l|}
+\hline
+Type & Offset & Size (bytes) & Content \\
+\hline \hline
+Request & 0 & 0 & no extra data \\
 \hline
 \end{tabular}
 
+The \msgref{EVENT_CONFIG0} message is used when the extra information in 
+\msgref{EVENT_CONFIG} is not available. In general \msgref{EVENT_CONFIG} 
+\emph{SHOULD} be prefered to \msgref{EVENT_CONFIG0} but this is not always
+possible. If the driver side bus implementation is manufacturing notification
+based on polling intervals, it should use this message to notifiy the driver
+side transport layer that it should use \msgref{GET_DEVICE_STATUS} and
+\msgref{GET_CONFIG} to query for device side changes.
+
+This message may also be used when a single configuration change on the device
+side cannot be communicated in one \msgref{EVENT_CONFIG} message due to maximum
+message size and does not desire to represent it as multiple configuration
+changes.
+
 \msgdef{EVENT_CONFIG}
 
-This message is sent by the virtio-msg backend and does not require a response.
+This message is sent by the virtio-msg device and there is no response.
```

Comment:
We might need to add something in here to mention that data might be empty (offset = num_bytes = 0)

Reply:
Sure, we can be more explicit.